src/ripple/app/tx/impl/InvariantCheck.h

```diff
@@ -168,6 +168,7 @@ using InvariantChecks = std::tuple<
  *
  * @see ripple::InvariantChecker_PROTOTYPE
  */
+template<class = void>
```

Comment:
There are two ways I know to solve the duplicate symbol issue you encountered: 1) The `template<class = void>` as above; 2) `inline`.

We've been using the `template<class = void>` when we would prefer that a function _not_ be inlined. For this function, it make sense to `inline` it. So I'd prefer `inline` in this case.

I think `inline` is more familiar to most C++ programmers, and we should prefer vanilla solutions unless we have a reason to reach for fancier idioms.

No change is needed for this case, but I'd rather not see `template<class = void>` replace `inline` in general.